### PR TITLE
Publisher author genre updatedbook model

### PR DESCRIPTION
**Pull request by hkim53**
We created the publisher, author, and genre models. We updated the associations on the book model. Please review.  We'd like to review the genre and the idea of 'unique' validations with y'all.

### CODE REVIEW

File: db/models/genre.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+'use strict'
+
+const bcrypt = require('bcryptjs')
+    , {STRING, VIRTUAL, TEXT, ENUM} = require('sequelize')
+
+module.exports = db => db.define('genre', {
+  genreType: {
+    type: ENUM,
+    values: ['fantasy', 'mystery', 'historical', 'sci-fi', 'romance', 'horror', 'action', 'death', 'thriller'],
+    isUnique:
```
Comment (sbakkila) [inline db/models/genre.js]: comment out broken code!

File: db/models/publisher.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+'use strict'
+
+const bcrypt = require('bcryptjs')
```
Comment (sbakkila) [inline db/models/publisher.js]: don't need bcrypt outside of the user model!

**Comment by dionysiaA** [comment]
@hkim53 @gooptaa one small detail, since you have created this new models, can you please require them in index.js so we won't face errors with the associations! 👍 🥇

File: db/models/genre.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    //     where: {
+    //       lastName
+    //     }
+    //   })
+    // }
+  }
+})
+
+module.exports.associations = (Genre, {Book}) => {
+    Genre.belongToMany(Book, {through: BookGenre})
```
Comment (sbakkila) [inline db/models/genre.js]: BooksGenres -- plurals for through associations

File: db/models/publisher.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+module.exports = db => db.define('publisher', {
+  name: {
+    type: STRING,
+    validate: {
+      notEmpty: true
+    }
+  },
+}, {
+  classMethods: {
```
Comment (sbakkila) [inline db/models/publisher.js]: delete empty classMethods

File: db/models/genre.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    , {STRING, VIRTUAL, TEXT, ENUM} = require('sequelize')
+
+module.exports = db => db.define('genre', {
+  genreType: {
+    type: ENUM,
+    values: ['fantasy', 'mystery', 'historical', 'sci-fi', 'romance', 'horror', 'action', 'death', 'thriller'],
+    isUnique: 
+  },
+}, {
+  classMethods: {
```
Comment (dionysiaA) [inline db/models/genre.js]: the same thing here! if its not being used then lets remove it! :)

**Pull request merged**